omod/src/main/java/org/openmrs/web/attribute/handler/RegexValidatedTextDatatypeHandler.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.attribute.handler;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.customdatatype.InvalidCustomValueException;
+import org.openmrs.customdatatype.datatype.RegexValidatedTextDatatype;
+import org.springframework.stereotype.Component;
+
+/**
+ * A handler for the {@link RegexValidatedTextDatatype}.
+ */
+@Component
+public class RegexValidatedTextDatatypeHandler implements FieldGenDatatypeHandler<RegexValidatedTextDatatype, java.lang.String> {
+	
+	
+	private static final Log log = LogFactory.getLog(RegexValidatedTextDatatypeHandler.class);
+	
+	/**
+	 * @see org.openmrs.customdatatype.CustomDatatypeHandler#setHandlerConfiguration(String)
+	 */
+	@Override
+	public void setHandlerConfiguration(String arg0) {
+		// not used
+	}
+	
+	/**
+	 * @see org.openmrs.web.attribute.handler.FieldGenDatatypeHandler#getWidgetName()
+	 */
+	@Override
+	public String getWidgetName() {
+		return "java.lang.String";
+	}
+	
+	/**
+	 * @see org.openmrs.web.attribute.handler.FieldGenDatatypeHandler#getWidgetConfiguration()
+	 */
+	@Override
+	public Map<String, Object> getWidgetConfiguration() {
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.web.attribute.handler.FieldGenDatatypeHandler#getValue(CustomDatatype,
+	 *      HttpServletRequest, String)
+	 */
+	@Override
+	public String getValue(RegexValidatedTextDatatype datatype, HttpServletRequest request, String formFieldName)
+	        throws InvalidCustomValueException {
+		String stringVal = request.getParameter(formFieldName);
+		try {
+			datatype.validate(stringVal);
```

Comment:
it is done by [DateFieldGenDatatypeHandler ](https://github.com/teleivo/openmrs-module-legacyui/blob/03e0c53d56b24c43e8304e56e53469a07da2946a/omod/src/main/java/org/openmrs/web/attribute/handler/DateFieldGenDatatypeHandler.java#L67-L69) thats where I got this from and its not necessarily wrong since I am allowed to throw a InvalidCustomValueException.

So currently it behaves like this when entering an invalid value according to the regex:
![image](https://cloud.githubusercontent.com/assets/4661144/16487118/206b13bc-3ec9-11e6-852f-141d858ad280.png)

it also replaces the invalid value with the latest saved GP which was valid.


If I change to

```
@Override
	public String getValue(RegexValidatedTextDatatype datatype, HttpServletRequest request, String formFieldName)
	        throws InvalidCustomValueException {
		return request.getParameter(formFieldName);
	}
```

it becomes:

![image](https://cloud.githubusercontent.com/assets/4661144/16521294/355afb34-3f96-11e6-9b34-818ff0e3c19f.png)

and leaves the invalid value.

@dkayiwa and @wluyima just tell me what you prefer and I can make another PR.